### PR TITLE
[NativeAOT-LLVM] For TARGET_WASM do not expand to GT_INDEX_ADDR

### PR DESCRIPTION
**Pull request by yowl**
This PR adds support in the LLVM clrjit for the `GT_INDEX_ADDR` instruction.  Do I need some todo's in here for 64 bit Wasm on either the index, the element offset (unlikely I suppose, its just the type + length I think), or the element size?

@cc SingleAccretion.

Thanks,

### CODE REVIEW

**Comment by yowl** [comment]
cc @SingleAccretion  (had @ in the wrong place)

File: src/coreclr/jit/llvm.cpp
Comment (SingleAccretion) [inline src/coreclr/jit/llvm.cpp]: Hmm, this won't work as-is, `INDEX_ADDR` is effectively bounds-checked (and we're missing the bounds check).

I think it would be better to just always fully expand `INDEX` in morph for LLVM, `INDEX_ADDR`'s only purpose is to increase throughput under MinOpts, and I think that sort of tuning can wait (and is probably not worthwhile anyway).

Comment (yowl) [reply src/coreclr/jit/llvm.cpp]: I see you've got a PR in runtime around this area.  Would you prefer this was left until that was accepted and merged here?

Comment (SingleAccretion) [reply src/coreclr/jit/llvm.cpp]: I don't think there is a need to block this change on my changes in the runtime.

The fix here should be very simple: `#ifdef`-out the logic that creates `INDEX_ADDR`s for `TARGET_WASM` in `fgMorphArrayIndex`.

Comment (yowl) [reply src/coreclr/jit/llvm.cpp]: thanks, have #if'ed it out.

**Review by SingleAccretion — APPROVED** [review]

**Pull request merged**
